gnomad_qc/v4/annotations/insilico_predictors.py

```diff
@@ -89,6 +90,131 @@ def _load_cadd_raw(cadd_tsv) -> hl.Table:
+
+    SpliceAI scores are from the following resources:
+        - Precomputed SNVs: spliceai_scores.masked.snv.hg38.vcf.bgz,
+          downloaded from https://basespace.illumina.com/s/5u6ThOblecrh
+        - Precomputed indels: spliceai_scores.masked.indel.hg38.vcf.bgz,
+          downloaded from https://basespace.illumina.com/s/5u6ThOblecrh
+        - gnomAD v4 indels: gnomad_v4_new_indels.spliceai_masked.vcf.bgz,
+          computed on v4 by Illumina.
+
+    :param str snvs_path: Path to the precomputed SNVs.
+    :param str indels_path: Path to the precomputed indels.
+    :param str new_indels_path: Path to the gnomAD v4 indels.
+    :return: Hail Table with SpliceAI scores for GRCh38.
+    """
+
+    def import_spliceai_vcf(path: str) -> hl.Table:
+        """
+        Import SpliceAI VCF into Hail Table.
+
+        :param str path: Path to SpliceAI VCF.
+        :return: Hail MatrixTable with SpliceAI scores.
+        """
+        ht = hl.import_vcf(
+            path,
+            force_bgz=True,
+            header_file=header_file_path,
+            reference_genome="GRCh38",
+            contig_recoding=NO_CHR_TO_CHR_CONTIG_RECODING,
+            skip_invalid_loci=True,
+            min_partitions=1000,
+        ).rows()
+        return ht
+
+    logger.info("Importing vcf of SpliceAI scores into MT...")
+
+    spliceai_snvs = import_spliceai_vcf(snvs_path)
+    spliceai_indels = import_spliceai_vcf(indels_path)
+    spliceai_new_indels = import_spliceai_vcf(new_indels_path)
+
+    ht = spliceai_snvs.union(spliceai_indels).union(spliceai_new_indels)
+
+    logger.info("Exploding SpliceAI scores...")
+    # `explode` will eliminate rows with empty array, some variants don't have
+    # score might because they are located inside gene body, but the variants
+    # fall on multiple genes will extend the number of rows.
+    ht = ht.explode(ht.info.SpliceAI)
+
+    logger.info("Annotating SpliceAI scores...")
+    # there will only be one gene in the array after exploding
+    gene_symbol = ht.info.SpliceAI.split(delim="\\|")[1:2][0]
+    # delta_score array for 4 splicing consequences: DS_AG|DS_AL|DS_DG|DS_DL
+    delta_scores = ht.info.SpliceAI.split(delim="\\|")[2:6]
+    # position array is |AG|AL|DG|DL
+    positions = ht.info.SpliceAI.split(delim="\\|")[6:10]
+    ht = ht.annotate(
+        splice_ai=hl.struct(
+            gene_symbol=gene_symbol,
+            delta_scores=hl.map(lambda x: hl.float32(x), delta_scores),
+            positions=hl.map(lambda x: hl.int32(x), positions),
+        )
+    )
+
+    # Annotate info.max_DS with the max of DS_AG, DS_AL, DS_DG, DS_DL in info.
+    logger.info(
+        "Getting the max SpliceAI score across consequences for each variant pergene..."
+    )
+    consequences = hl.literal(
+        ["acceptor_gain", "acceptor_loss", "donor_gain", "donor_loss"]
+    )
+    ht = ht.annotate(
+        splice_ai=ht.splice_ai.annotate(ds_max=hl.max(ht.splice_ai.delta_scores))
+    )
```

Comment:
```suggestion
    ht = ht.annotate(delta_scores=hl.map(lambda x: hl.float32(x), delta_scores))
     # Annotate ds_max with the max of DS_AG, DS_AL, DS_DG, DS_DL in info.
    logger.info(
        "Getting the max SpliceAI score across consequences for each variant per"
        " gene..."
    )
    ht = ht.select(ds_max=hl.max(ht.delta_scores))
    )
```
I don't think you need the struct just yet since we arent using those fields and it's less easier to handle the single top level annotation in the code rather than a single nested annotation.